docs/docs/how-to/plugins-and-themes/creating-a-source-plugin.md

```diff
@@ -806,6 +806,167 @@ One challenge when developing locally is that a developer might make modificatio
+
+```javascript:title=source-plugin/utils.js
+// highlight-start
+let warnOnceForNoSupport = false
+// highlight-end
+
+export function createNodeManifest({
+  // ...
+}) {
+  // ...
+
+  if (shouldCreateNodeManifest) {
+    // highlight-start
+    const updatedAt = entryItem.updatedAt
+    const manifestId = `${project.id}-${entryItem.id}-${updatedAt}`
+
+    unstable_createNodeManifest({
+      manifestId,
+      node: entryNode,
+      updatedAtUTC: updatedAt,
+    })
+  }
+}
+```
+
+##### Warn if no support
+
+Lastly we'll want to give our users a good experience and give a warning if they're using a version of Gatsby that does not support Content Sync
+
+```javascript:title=source-plugin/utils.js
+// highlight-start
+let warnOnceForNoSupport = false
+// highlight-end
+
+export function createNodeManifest({
+  // ...
+}) {
+  // ...
+
+  if (shouldCreateNodeManifest) {
+    // ...
+  } else if (
+    // highlight-start
+    // it's helpful to let users know if they're using an outdated Gatsby version so they'll upgrade for the best experience
+    isPreview && !createNodeManifestIsSupported && !warnOnceForNoSupport
+  ) {
+    console.warn(
+      `${sourcePluginName}: Your version of Gatsby core doesn't support Content Sync (via the unstable_createNodeManifest action). Please upgrade to the latest version to use Content Sync in your site.`
+    )
+    // This is getting called for every entry node so we don't want the console logs to get cluttered
+    warnOnceForNoSupport = true
+    // highlight-end
+  }
+}
+```
+
+#### Content Sync data source (or CMS) changes
+
+The CMS will need to send a webhook when content is changed
+
+##### Using the gatsby preview extension
+
+This package is currently in the process of being open sourced and should be available shortly. For context, here is what the package provides, a button to handle opening the Content Sync waiting room. You will need to create a button similar to this for the time being.
+![Open Preview Button Screenshot](https://user-images.githubusercontent.com/18426780/61498855-37e35a00-a982-11e9-8201-ab96be74f1f1.png)
+
+##### Configuration
+
+You will need to store the Content Sync URL from a given Gatsby Cloud site. This will look something like `https://gatsbyjs.com/content-sync/<siteId>`. This is often done in the plugin extension or app extension configuration, this will differ from CMS to CMS depending on how they handle their plugin ecosystem.
+
+##### Building up the manifest id
+
+Recall that we need to create a matching manifest id in the CMS AND the Gatsby plugin. Whenever content is saved we can build up a new manifest id that will look the same as the manifest id we created in the source plugin
+
+```javascript
+const manifestId = `${project.id}-${entryItem.id}-${updatedAt}`
```

Comment:
A small suggestion to make this a little more generic. `entryItem` is Contentful specific and also might imply some sort of top level node in a tree which might be a tad confusing if you aren't thinking about Contentful already :slightly_smiling_face: 

```suggestion
const manifestId = `${project.id}-${blogPost.id}-${updatedAt}`
```

Reply:
I already explain the context of an entryItem earlier when I talk through source plugins and the idea of an entry node is used repeatedly throughout, this is code copied directly from there. This was to demonstrate that it will be the exact same code in both source plugin and CMS to build up the manifestId

> The first thing you'll want to do is identify which nodes you'll want to create a node manifest for. These will typically be nodes that you can preview, entry nodes, top level nodes, etc. An example of this could be a blog post or an article, any node that can be the "owner" of a page. A good place to call this action is whenever you call `createNode`.

Reply:
Aha, got it. It makes sense to leave as is :white_check_mark: